php/libraries/SiteIDGenerator.php

```diff
@@ -44,54 +44,88 @@ class SiteIDGenerator extends IdentifierGenerator
      */
     public function __construct(string $siteAlias, string $projectAlias)
     {
-        // Read config settings from project/config.xml to retrieve the
-        // alphabet, length, and generation method (sequential or random) used
-        // to create new IDs.
-        $this->generationMethod = $this->_getIDSetting('generation');
-        $this->length           = $this->_getIDSetting('length') ?? self::LENGTH;
-        $this->alphabet         = $this->_getIDSetting('alphabet');
+        $config = \NDB_Factory::singleton()->config();
+        // Get alphabet, length, and generation method (sequential or random)
+        // used to create new IDs.
+        $this->generationMethod = $config->getSetting('idGenerationMethod');
+        $configLength           = intval($config->getSetting('idLength'));
+        // Length should not be greater than the value of the constant.
+        $this->length = (self::LENGTH < $configLength) ?
+            self::LENGTH
+            : $configLength;
+
+        // get alphabet setting
+        switch ($config->getSetting('idAlphabet')) {
+        case 'alpha':
+            $this->alphabet = range('A', 'Z');
+            break;
+        case 'numeric':
+            $this->alphabet = range('0', '9');
+            break;
+        case 'alphanumeric':
+            $this->alphabet = array_merge(range('0', '9'), range('A', 'Z'));
+            break;
+        }
+
         // Initialize minimum and maximum allowed values for IDs. Set the values
         // to the lowest/highest character in $alphabet repeated $length times
-        // if the min or max is not configured in project/config.xml
-        $this->minValue = $this->_getIDSetting('min') ??
-            str_repeat(strval($this->alphabet[0]), $this->length);
-        $this->maxValue = $this->_getIDSetting('max') ??
+        // if the min or max is not configured in the Config table.
+        $this->minValue = $config->getSetting('idMinValue') ??
+            str_repeat(
+                strval($this->alphabet[0]),
+                $this->length
+            );
+        $this->maxValue = $config->getSetting('idMaxValue') ??
             str_repeat(
             strval($this->alphabet[count($this->alphabet) - 1]),
             $this->length
         );
-
         $this->siteAlias    = $siteAlias;
         $this->projectAlias = $projectAlias;
-        $this->prefix       = $this->_getIDSetting('prefix');
+        // Determine prefix to be used for the ID.
+        switch ($config->getSetting('idPrefixMethod')) {
+        case 'static':
+            $this->prefix = $config->getSetting('idStaticPrefix');
+            break;
+        case 'alias':
+            $this->prefix = $siteAbbrevPrefix;
```

Comment:
i think this was not modified to account for projects

Reply:
Yeah I think this PR predates that change. It'll need to be updated